transport/http/inbound.go

```diff
@@ -68,7 +69,13 @@ type Inbound struct {
 	server     *intnet.HTTPServer
 	registry   transport.Registry
 	tracer     opentracing.Tracer
-	running    atomic.Bool
+
+	started   atomic.Bool
+	startOnce sync.Once
+	startErr  error
```

Comment:
Avoid repeating all this logic so many times.

Create a single type that encapsulates the "`sync.Once` with support for
errors and querying" logic:

```go
type Once struct {
    done atomic.Bool
    once sync.Once
    err error
}

func (o *Once) Do(func() error) error {
    // ...
}

func (o *Once) Done() bool {
    // ...
}
```

Then use it wherever you have that triple:

```go
type Inbound struct {
    startOnce, stopOnce Once
}

func (i *Inbound) Start() error {
    return i.startOnce.Do(i.start)
}

func (i *Inbound) Stop() error {
    return i.stopOnce.Do(i.stop)
}

func (i *Inbound) IsRunning() bool {
    return i.startOnce.Done() && !i.stopOnce.Done()
}
```

Optionally, you could also pull this logic out into a separate type to DRY
this up further.

```go
type LifecycleOnce struct {
    start Once
    stop  Once
}

func (l *LifecycleOnce) Start(f func() error) error {
    return l.start.Do(f)
}

func (l *LifecycleOnce) Stop(f func() error) error {
    return l.stop.Do(f)
}

func (l *LifecycleOnce) IsRunning() bool {
    return l.start.Done() && !l.stop.Done()
}
```

Then,

```go
type Inbound struct {
    once LifecycleOnce
    // ..
}

func (i *Inbound) Start() error {
    return i.once.Start(i.start)
}

func (i *Inbound) Stop() error {
    return i.once.Stop(i.stop)
}

func (i *Inbound) IsRunning() bool {
    return i.once.IsRunning()
}
```